controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -76,6 +77,169 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// PodCIDR
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	PodCIDR string `json:"PodCIDR,omitempty"`
+
+	// ServiceCIDR
+	// +kubebuilder:validation:Format=cidr
+	// +optional
+	ServiceCIDR string `json:"serviceCIDR,omitempty"`
+
+	// HostPrefix
+	// +optional
+	HostPrefix int `json:"hostPrefix,omitempty"`
+
+	// The compute machine type to use, for example `r5.xlarge`.
+	// +optional
+	ComputeMachineType RosaMachineType `json:"computeMachineType,omitempty"`
+
+	// Cluster AutoScaler.
+	// +optional
+	ClusterAutoScaler ClusterAutoscaler `json:"clusterAutoScaler,omitempty"`
```

Comment:
we only need `AutoScaling` with Min and Max replicas exactly like https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/api/v1beta2/rosamachinepool_types.go#L60
 
 
clusterAutoscaler is a different thing and is not supported for hcp.

Reply:
you can just import the `RosaMachinePoolAutoScaling` type https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/api/v1beta2/rosamachinepool_types.go#L73-L78 and use it here.

Reply:
done